Feature/#34 [OrderProduct] 주문상품 생성, 주문과 외래키(연관관계) 설정

jin2304 opened the pull request:
## #️⃣연관된 이슈
- 이슈 번호: #34 

## 변경 타입
- [x] 신규 기능 추가/수정

## 변경 내용
- **to-be**(변경 후 설명을 여기에 작성)
 -주문상품 entity 추가
 -주문상품 테이블 생성 및 주문 테이블과 외래키 설정

## 체크리스트
- [x] 코드가 제대로 동작하는지 확인했습니다.

## 코멘트
- @OneToMany로 설계했을 때 INSERT 발생 시 추가적인 UPDATE가 필요다는 점과 Entity에서 외래키를 매핑하는 필드가 보이지 않아 파악하기 어려운 단점이 있어 @ManyToOne 으로 설계
- 향후 Order에서 OrderProduct를 조회하는 경우가 필요할 때 양방향 고려

jin2304 committed:
[#34] feat: 주문상품 생성, 주문과 외래키(연관관계) 설정

-주문상품 entity 추가
-주문상품 테이블 생성 및 주문 테이블과 외래키 설정

je-pa commented on src/main/java/com/toyland/order/model/Order.java:
```diff
@@ -28,4 +31,11 @@ public class Order {
     @Column(name = "payment_Type")
     private PaymentType paymentType; // 결제유형(카드/현금)
+
+    @OneToMany
+    @JoinColumn(name = "order_id") // p_order_product 테이블에 order_id 컬럼으로 외래키 설정
+    private List<OrderProduct> orderProductList = new ArrayList<>();
```
보통 @OneToMany 단방향보단 @ManyToOne이 많이 사용되는걸로 알고있는데 @OneToMany를 선택하신 이유가 있으실까요?

jin2304 replied:
앗 이유는 딱히 없었는데, 찾아보니 @OneToMany로 설계했을 때  INSERT 발생 시 추가적인 UPDATE가 필요다는 점과 Entity에서 외래키를 매핑하는 필드가 보이지 않아 파악하기 어려운 단점이 있어 @ManyToOne 으로 설계하는게 좋은 방법인 것 같네요! 수정하겠습니다.

je-pa commented on src/main/java/com/toyland/orderproduct/model/OrderProduct.java:
```diff
@@ -0,0 +1,29 @@
+    @Column(name = "order_product_price", updatable = false, nullable = false)
+    private String orderProductPrice; // 주문 가격(주문 당시 가격)
+
+    @Column(name = "content", nullable = false)
+    private String quantity;
```
변수명은 quantity인데 name은 content네요!

jin2304 replied:
감사합니다. 수정하겠습니다!

je-pa reviewed (commented):
고생하셨습니다!

hanjihoon03 commented:
고생하셨어요! 리뷰 보고 답변 변경 해주세요!

jin2304 committed:
[#34] refactor: @OneToMany -> @ManyToOne 방식으로 수정

-@OneToMany로 설계했을 때 INSERT 발생 시 추가적인 UPDATE가 필요다는 점과 Entity에서 외래키를 매핑하는 필드가 보이지 않아 파악하기 어려운 단점이 있어 @ManyToOne 으로 설계

jin2304 committed:
[#34] refactor: OrderProduct 엔티티 필드명 수정

jin2304 committed:
[#34] refactor: @OneToMany -> @ManyToOne 방식으로 수정2

-@OneToMany로 설계했을 때 INSERT 발생 시 추가적인 UPDATE가 필요다는 점과 Entity에서 외래키를 매핑하는 필드가 보이지 않아 파악하기 어려운 단점이 있어 @ManyToOne 으로 설계

jin2304 commented:
> 고생하셨어요! 리뷰 보고 답변 변경 해주세요!

수정완료했습니다!
